Index: basicstation-2.0.3/src/aio.c
===================================================================
--- basicstation-2.0.3.orig/src/aio.c
+++ basicstation-2.0.3/src/aio.c
@@ -33,6 +33,7 @@
 #include <fcntl.h>
 #include "rt.h"
 
+#include <sys/select.h>
 
 enum { N_AIO_HANDLES = 10 };
 static aio_t aioHandles[N_AIO_HANDLES];
Index: basicstation-2.0.3/CMakeLists.txt
===================================================================
--- basicstation-2.0.3.orig/CMakeLists.txt
+++ basicstation-2.0.3/CMakeLists.txt
@@ -116,8 +116,9 @@ target_compile_options(
 	-DCFG_no_leds
 	-DCFG_lgw1
 	-DCFG_logini_lvl=INFO
-	-DCFG_tlsdebug
+#	-DCFG_tlsdebug
 	-DCFG_ral_lgw
+	-DCFG_argp
 )
 
 target_compile_definitions(
