Index: lora_gateway-5.0.1/libloragw/src/loragw_spi.native.c
===================================================================
--- lora_gateway-5.0.1.orig/libloragw/src/loragw_spi.native.c
+++ lora_gateway-5.0.1/libloragw/src/loragw_spi.native.c
@@ -53,7 +53,11 @@ Maintainer: Sylvain Miermont
 
 #define READ_ACCESS     0x00
 #define WRITE_ACCESS    0x80
+
+#ifndef SPI_SPEED
 #define SPI_SPEED       8000000
+#endif
+
 #ifndef SPI_DEV_PATH
 #define SPI_DEV_PATH    "/dev/spidev0.0"
 #endif
