Index: lorawan-stack-3.9.0/tools/mage/git.go
===================================================================
--- lorawan-stack-3.9.0.orig/tools/mage/git.go
+++ lorawan-stack-3.9.0/tools/mage/git.go
@@ -62,7 +62,7 @@ func (g Git) InstallHooks() error {
 }
 
 func init() {
-	initDeps = append(initDeps, Git.InstallHooks)
+//	initDeps = append(initDeps, Git.InstallHooks)
 }
 
 // UninstallHooks uninstalls git hooks.
