Index: lorawan-stack-3.3.1/.mage/git.go
===================================================================
--- lorawan-stack-3.3.1.orig/.mage/git.go
+++ lorawan-stack-3.3.1/.mage/git.go
@@ -61,7 +61,7 @@ func (g Git) InstallHooks() error {
 }
 
 func init() {
-	initDeps = append(initDeps, Git.InstallHooks)
+//	initDeps = append(initDeps, Git.InstallHooks)
 }
 
 // UninstallHooks uninstalls git hooks.
